File: tests/07_phases_ompss_2.dg/fortran/success_lint_02.f90
! <testinfo>
! test_generator=config/mercurium-ompss-2
! test_nolink=yes
! </testinfo>
PROGRAM P
    IMPLICIT NONE
    INTEGER, ALLOCATABLE :: V(:)
    INTEGER, PARAMETER ::  N = 100

    !$OSS LINT ALLOC(V(1:N))
        ALLOCATE(V(N))
    !$OSS END LINT

    !$OSS LINT FREE(V)
        DEALLOCATE(V)
    !$OSS END LINT
END PROGRAM P

File: tests/07_phases_ompss_2.dg/c/success_lint_02.c
/*
<testinfo>
test_generator=config/mercurium-ompss-2
test_nolink=yes
</testinfo>
*/
int main()
{
    int *p = 0;
    int N = 100;
    #pragma oss lint alloc(p[0;N])
    {
        //p = malloc(N*sizeof(*p));
    }

    #pragma oss lint free(p)
    {
        // free(p);
    }
}
